api/hasura/cron/epochs.ts

```diff
@@ -0,0 +1,348 @@
+              users_aggregate: [
+                { where: { non_giver: { _eq: false } } },
+                {
+                  aggregate: { count: [{}, true] },
+                },
+              ],
+            },
+          },
+        ],
+      },
+      notifyEnd: {
+        epochs: [
+          {
+            where: {
+              _and: [
+                { end_date: { _gt: 'now()' } },
+                { end_date: { _lte: inTwentyFourHours } },
+              ],
+              start_date: { _lt: 'now()' },
+              notified_before_end: { _is_null: true },
+            },
+          },
+          {
+            id: true,
+            circle_id: true,
+            number: true,
+            end_date: true,
+            circle: {
+              id: true,
+              name: true,
+              telegram_id: true,
+              discord_webhook: true,
+              organization: { name: true },
+              users: [
+                {
+                  where: {
+                    non_giver: { _eq: false },
+                    give_token_remaining: { _gt: 0 },
+                  },
+                },
+                {
+                  name: true,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      endEpoch: {
+        epochs: [
+          {
+            where: {
+              end_date: { _lt: 'now()' },
+              notified_end: { _is_null: true },
+            },
+          },
+          {
+            repeat: true,
+            repeat_day_of_month: true,
+            circle: {
+              telegram_id: true,
+              discord_webhook: true,
+              users: [{}, { circle_id: true, bio: true }],
+            },
+            epoch_pending_token_gifts: [
+              {},
+              {
+                circle_id: true,
+                epoch_id: true,
+                note: true,
+                sender_id: true,
+                sender_address: true,
+                recipient_id: true,
+                recipient_address: true,
+              },
+            ],
+          },
+        ],
+      },
+    },
+  });
+  return result;
+}
+
+export async function notifyEpochStart({
+  notifyStart: { epochs },
+}: EpochsToNotify) {
+  const sendNotifications = epochs.map(async epoch => {
+    const { start_date, end_date, circle, number: epochNumber } = epoch;
+
+    if (epochNumber == null) await setNextEpochNumber(epoch);
+
+    assert(circle, 'panic: no circle for epoch');
+    const epochStartDate = DateTime.fromISO(start_date);
+    const epochEndDate = DateTime.fromISO(end_date);
+    const eligibleUsersCount = circle.users_aggregate.aggregate?.count;
+
+    const message = dedent`
+      A new ${circle.organization?.name}/${circle.name} epoch is active!
+      ${eligibleUsersCount} users will be participating and the duration of the epoch will be:
+      **${epochStartDate.toLocaleString(
+        DateTime.DATETIME_FULL
```

Comment:
maybe we can add this formatting into the `dateTimeHelper.ts` , so the date format is consistent. `DateTime.DATETIME_FULL` format actually looks better, we can replacing the existing one with that.